lib/taurus/qt/qtgui/input/tauruslineedit.py

```diff
@@ -124,15 +128,21 @@ def isTextValid(self):
     def updateStyle(self):
         TaurusBaseWritableWidget.updateStyle(self)
 
-        value = self.getValue()
-        if value is None and self.hasPendingOperations():
+        if self._last_value is None and self.hasPendingOperations():
+            # This check must be done at updateStyle() because pending
+            # operations list is not updated yet at handleEvent() stage
             try:
+                self.getModelObj().getAttributeInfoEx(cache = False)
                 value = self.getModelValueObj().wvalue
+                self.info('Overwriting wvalue=None with %s' % (value))
                 self.setValue(value)
```

Comment:
I think that updateStyle is not the right place for setting the value because:

- it may be called very often
- it is likely to produce bugs such as those seen in #650

Reply:
IMHO, we really should try to think of a way to do this kind of stuff in handleEvent